### DISCUSSION TITLE
How can I disable ale signs/highlights while I'm in insert mode?

### DISCUSSION THREAD

**Discussion by ayushnix** [w0rp/ale#3836] [Q&A - Ask for help with problems]
I want to disable all the signs/highlights in the sidebar and the status bar at the bottom while I'm in insert mode/whenever I'm typing something. When I switch to normal mode, I want to see all the signs again.

Basically, it should be as if ale isn't installed whenever I'm tying something or when I'm in insert mode. However, when I'm normal mode, I want to see everything again.

I did use the following options but they still show me error/warning signs when I'm in insert mode.

```
let g:ale_lint_on_text_changed = 'normal'
let g:ale_lint_on_insert_leave = 1
```

**Comment by ayushnix** [comment]
So I managed to implement a solution using NeoVim `autocmd`

```
augroup disable_ale_insmode
    autocmd!
    autocmd InsertEnter * :ALEToggle
    autocmd InsertLeave * :ALEToggle
augroup END
```

I guess we can also use `ALEDisableBuffer` and `ALEEnableBuffer` instead of `ALEToggle`.

**Reply by ayushnix** [reply]
This solution doesn't work if one wants to leave the sign column always enabled to prevent shift of text.

**Comment by hsanson** [comment]
There is no command or function in ALE to disable/toggle highlighting or signs.  This is a new feature and you can create an issue to request it or implement it yourself.

I have not tried this but you may be able to create a function that modifies the ALE signs and highlights so they look disabled when you enter insert mode and another function that restores them when you enter normal mode. For example setting ALEError, ALEWarning, ALEInfo, ALEStyleError, ALEStyleWarning to a blank space and the highlight groups to your editor defaults:

- ALESignColumnWithErrors
- ALESignColumnWithoutErrors
- ALEErrorLine
- ALEWarningLine
- ALEInfoLine

There may be many others so check the documentation for highlight groups used by ALE.
